sabnzbd/downloader.py

```diff
@@ -532,21 +548,35 @@ def run(self):
                             server.request_info()
                         break
 
-                    article = sabnzbd.NzbQueue.get_article(server, self.servers)
+                    try:
+                        if server.article_queue[0].nzf.nzo.status == Status.PAUSED:
```

Comment:
I don't think this logic should be here. It's part of the pause-event to clear the queue, so maybe this should be part of `nzo.pause`.

Reply:
I'm not sure how well that would work if I understand you correctly. `nzo.pause` would have to go through all the server queues and look for articles belonging to that nzo, and there's the issue of locking. It could find an article, then the pause is called which clears the queue, and then that article is used to add 4 more articles to the queue. All those articles would be downloaded before it notices the pause.

Reply:
Then I think we can just remove this check. It's just not the place for it. Worst case it downloads 4 articles, which is like 750kb * 4.
It also doesn't account for changing the order of the queue for example, or setting job priority to Force and then back to normal etc etc. That's all fine for 4 articles.

Reply:
Unfortunately it would keep reusing articles from that file to download more until it was fully downloaded. I've changed that so that now it will add (threads / 5) articles once instead. That should reduce the problem with downloading articles.

The main advantages of the patch are a hopefully better retries system and much faster marking of expired articles as tried.